ui/react-components/components/EditAppointment/EditAppointment.jsx

```diff
@@ -0,0 +1,255 @@
+import CalendarPicker from "../CalendarPicker/CalendarPicker.jsx";
+import AppointmentDatePicker from "../DatePicker/DatePicker.jsx";
+
+const EditAppointment = props => {
+
+    const {appConfig, appointmentUuid, isRecurring} = props;
+
+    const [errors, setErrors] = useState({
+        patientError: false,
+        serviceError: false,
+        appointmentDateError: false,
+        startDateError: false,
+        endDateError: false,
+        endDateTypeError: false,
+        occurrencesError: false,
+        startTimeError: false,
+        endTimeError: false,
+        recurrencePeriodError: false,
+        startTimeBeforeEndTimeError: false,
+        weekDaysError: false
+    });
+
+    const initialAppointmentState = {
+        patient: undefined,
+        providers: [],
+        service: undefined,
+        serviceType: undefined,
+        location: undefined,
+        speciality: undefined,
+        appointmentDate: undefined,
+        recurringStartDate: undefined,
+        recurringEndDate: undefined,
+        startTime: undefined,
+        endTime: undefined,
+        isRecurring: isRecurring,
+        notes: undefined,
+        recurrenceType: undefined,
+        occurrences: undefined,
+        period: undefined,
+        weekDays: undefined
+    };
+    const [appointmentDetails, setAppointmentDetails] = useState(initialAppointmentState);
+
+    const updateErrorIndicators = errorIndicators => setErrors(prevErrors => {
+        return {...prevErrors, ...errorIndicators}
+    });
+
+    const updateAppointmentDetails = modifiedAppointmentDetails => setAppointmentDetails(prevAppointmentDetails => {
+        return {...prevAppointmentDetails, ...modifiedAppointmentDetails}
+    });
+
+    //TODO To be checked if can be moved to common place
+    const endTimeBasedOnService = (time, service, serviceType) => {
+        const currentTime = moment(time);
+        const duration = getDuration(service, serviceType);
+        currentTime.add(duration, MINUTES);
+        if (time) {
+            updateAppointmentDetails({endTime: currentTime});
+            updateErrorIndicators({endTimeError: false});
+        }
+    };
+
+    const [currentStartTime, setCurrentStartTime] = useState();
+    const [currentEndTime, setCurrentEndTime] = useState();
+
+    const generateAppointmentDetails = async () => {
+        const appointment = isRecurring ? await getRecurringAppointmentByUuid(appointmentUuid) : await getAppointmentByUuid(appointmentUuid);
+        const appointmentResponse = isRecurring
+            ? (appointment && appointment.data && appointment.data.appointmentDefaultResponse) || undefined
+            : (appointment && appointment.data && appointment.data) || undefined;
+        const recurringPattern = isRecurring
+            ? (appointment && appointment.data && appointment.data.recurringPattern) || undefined : undefined;
+        if (appointmentResponse) {
+            updateAppointmentDetails({
```

Comment:
Can we move all the appointment specific things like these to an Appointment Model?? Talk to Alekhya and Siva also about this before doing anything.

Reply:
Will be handled later together for add and edit